File: typescript/src/models/cardFundingType.ts
// this file is @generated

export enum CardFundingType {
  Credit = "credit",
  Debit = "debit",
  Prepaid = "prepaid",
}

export const CardFundingTypeSerializer = {
  _fromJsonObject(object: any): CardFundingType {
    return object;
  },

  _toJsonObject(self: CardFundingType): any {
    return self;
  },
};

File: rust/src/models/card_funding_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum CardFundingType {
    #[serde(rename = "credit")]
    Credit,
    #[serde(rename = "debit")]
    Debit,
    #[serde(rename = "prepaid")]
    Prepaid,
}

impl fmt::Display for CardFundingType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::Credit => "credit",
            Self::Debit => "debit",
            Self::Prepaid => "prepaid",
        };
        f.write_str(value)
    }
}
